drivers/src/soc_ifc.rs

```diff
@@ -20,9 +20,23 @@ use crate::FuseBank;
 
 pub type Lifecycle = DeviceLifecycleE;
 
+// [TODO] Move memory layout out of the common crate
```

Comment:
to work around the circular dep issues, I think [memory_layout.rs](http://memory_layout.rs/) should just be in drivers/src/

Reply:
The problem with these sorts of TODOs is that they often get missed, and then we end up with hard-to-debug memory issues when the addresses change.

Reply:
I'll be addressing this TODO as part of https://github.com/chipsalliance/caliptra-sw/issues/609